### PR TITLE
 🏈 refactor move offer related concerns from application choice to the offer class

### PR DESCRIPTION
**Pull request by despo**
## Context

Move offer related content to `Offer` model by using has through associations and `delegate`

## Link to Trello card

https://trello.com/c/jEnBoj4Y/3721-%F0%9F%8F%88-refactor-move-offer-related-concerns-from-application-choice-to-the-offer-class

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: spec/models/offer_spec.rb [hunk 1]
```diff
@@ -9,5 +9,59 @@
 
       expect(offer.conditions.map(&:text)).to contain_exactly('Provide evidence of degree qualification', 'Do a backflip and send us a video')
     end
+
+    describe '#course_option' do
```
Comment (SamJamCul) [inline spec/models/offer_spec.rb:13]: Do we need to test model associations?

File: spec/models/offer_spec.rb [hunk 2]
```diff
@@ -9,5 +9,59 @@
...
+
+  describe '#unconditional' do
+    it 'returns true when there are no conditions' do
+      offer = create(:unconditional_offer)
+
+      expect(offer.unconditional?).to be true
+    end
+  end
+
+  context 'delegators' do
```
Comment (SamJamCul) [inline spec/models/offer_spec.rb:31]: Similarly, do we need to test delegations?

**Review by SamJamCul — COMMENTED** [review]
Looks good, I had a question about `unconditional_offer`, but that's about it!

File: app/models/offer.rb [hunk 3]
```diff
@@ -1,4 +1,17 @@
 class Offer < ApplicationRecord
   belongs_to :application_choice
   has_many :conditions, class_name: 'OfferCondition', dependent: :destroy
+
+  has_one :course_option, through: :application_choice, source: :current_course_option
+
+  delegate :course, :site, :provider, :accredited_provider, to: :course_option
+  delegate :offered_at, to: :application_choice
+
+  def unconditional?
```
[the remark this answers is not in the document]
Comment (despo) [reply app/models/offer.rb:10]: Yes, this is how we identify unconditional offers. In `application_choice` the method was used to indicate a state though, which caused some confusion.

File: spec/models/offer_spec.rb (hunk 1 above)
Comment (despo) [reply spec/models/offer_spec.rb:13]: I think in this instance we should, as the spec also acts as documentation for the class and the association is through another model.

File: spec/models/offer_spec.rb (hunk 2 above)
Comment (despo) [reply spec/models/offer_spec.rb:31]: Similar to my comment above, I think yes as the spec also acts as documentation for the model

**Review by SamJamCul — APPROVED** [review]
LGTM!

**Review by benswannack — APPROVED** [review]
Nice!

**Pull request merged**